### PR TITLE
Js 35 fix tests

### PR DESCRIPTION
**Pull request by dimkl**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Improve test coverage for `Clerk.setActive` and fix some flaky tests and typos. For more information check commit titles and descriptions.

### CODE REVIEW

File: packages/clerk-js/src/core/clerk.test.ts [hunk 1]
```diff
@@ -112,13 +109,23 @@ describe('Clerk singleton', () => {
...
+      cookieSpy = jest.spyOn(AuthenticationService.prototype, 'setAuthCookiesFromSession');
+    });
+
+    afterEach(() => {
       mockSession.remove.mockReset();
       mockSession.touch.mockReset();
+
+      cookieSpy?.mockRestore();
+      // cleanup global window pollution
+      (window as any).__unstable__onBeforeSetActive = null;
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/core/clerk.test.ts:124]: Oops, thanks :)

**Review by nikosdouvlis — APPROVED** [review]
Nicely done.

I noticed a couple of todo comments, are these for future actions?

File: packages/clerk-js/src/core/clerk.test.ts [hunk 2]
```diff
@@ -182,6 +192,80 @@ describe('Clerk singleton', () => {
...
+      });
+      const beforeEmitMock = jest.fn().mockImplementationOnce(() => {
+        executionOrder.push('before emit');
+        return Promise.resolve();
+      });
+
+      await sut.setActive({ session: mockSession2 as any as ActiveSessionResource, beforeEmit: beforeEmitMock });
+
+      await waitFor(() => {
+        expect(executionOrder).toEqual(['session.touch', 'set cookie', 'before emit']);
```
Comment (anagstef) [inline packages/clerk-js/src/core/clerk.test.ts:228]: @dimkl Nice one! 😃

**Review by anagstef — APPROVED** [review]
💯

**Pull request merged**